src/kerberos/mod.rs

```diff
@@ -944,7 +944,11 @@ impl<'a> Kerberos {
                     .ok_or_else(|| Error::new(ErrorKind::InvalidToken, "Input buffers must be specified"))?;
                 let input_token = SecurityBuffer::find_buffer(input, SecurityBufferType::Token)?;
 
-                let neg_token_targ: NegTokenTarg1 = picky_asn1_der::from_bytes(&input_token.buffer)?;
+                let neg_token_targ = {
+                    let mut d = picky_asn1_der::Deserializer::new_from_bytes(&input_token.buffer);
+                    let neg_token_targ: NegTokenTarg1 = KrbResult::deserialize(&mut d)??;
+                    neg_token_targ
+                };
```

Comment:
suggestion: This seems identical to this
```rust
let neg_token_targ: NegTokenTarg1 = picky_asn1_der::from_bytes::<KrbResult>(&input_token.buffer)??;
```
or this
```rust
let krb_result: KrbResult = picky_asn1_der::from_bytes::(&input_token.buffer)?;
let neg_token_targ: NegTokenTarg1 = krb_result?;
```

Reply:
It does not work:

![image](https://github.com/Devolutions/sspi-rs/assets/43034350/f0f1b3ab-e045-4107-a784-d00b38bb425a)

It's important to call [the custom `Rusult` deserialization](https://github.com/Devolutions/picky-rs/blob/master/picky-krb/src/data_types.rs#L384-L419) instead of the default one which is used for enums